tardis/visualization/widgets/abundance_vs_velocity.py

```diff
@@ -0,0 +1,264 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+import plotly.graph_objects as go
+
+
+def plot_abundance_vs_velocity(sim):
+    """
+    Plots the abundance of elements vs velocity provided simulation result using matplotlib.
+
+    Parameters
+    ----------
+    sim : Model Object containing
+
+        config : str or dict
+            filename of configuration yaml file or dictionary
+
+        atom_data : str or tardis.atomic.AtomData
+            if atom_data is a string it is interpreted as a path to a file storing
+            the atomic data. Atomic data to use for this TARDIS simulation.
+            If set to None, the atomic data will be loaded according to keywords set in the configuration
+            [default=None]
+
+
+    Returns
+    -------
+    None
+
+    """
+    # code goes here
+
+    # loading abundance sata in 'df'
+    df = (sim.model.abundance).T
+
+    # saving the velocity array in 'velocity'
+    velocity = sim.model.velocity
+
+    # obtaing a map of atomic numbers to element symbol
+    # the index of a will be the atomic number and corresponding value will be element symbol
+    a = atomic_number_to_element_symbol()
+
+    # plotting the graph of abundance of element vs velocity
+
+    # for loop iterate and plot abundance of each element at a time
+    for index, row in df.T.iterrows():
+        plt.plot(velocity[:-1], df[index], "o-", label=a[index])
+    plt.legend()
+    plt.title("Abundance of elements v/s Velocity")
+    plt.xlabel("Velocity in " + str(sim.model.velocity.unit))
+    plt.ylabel("Abundance (in Fraction)")
+    plt.show()
+
+
+def plotly_abundance_vs_velocity(sim):
+    """
+    Plots the abundance of elements vs velocity provided simulation result using plotly.
+
+    Parameters
+    ----------
+    sim : Model Object containing
+
+        config : str or dict
+            filename of configuration yaml file or dictionary
+
+        atom_data : str or tardis.atomic.AtomData
+            if atom_data is a string it is interpreted as a path to a file storing
+            the atomic data. Atomic data to use for this TARDIS simulation.
+            If set to None, the atomic data will be loaded according to keywords set in the configuration
+            [default=None]
+
+
+    Returns
+    -------
+    None
+
+    """
+
+    # loading abundance sata in 'df'
+    df = (sim.model.abundance).T
+
+    # saving the velocity array in 'velocity'
+    velocity = sim.model.velocity
+
+    # obtaing a map of atomic numbers to element symbol
+    # the index of a will be the atomic number and corresponding value will be element symbol
+    a = atomic_number_to_element_symbol()
+
+    # plotting the graph of abundance of element vs velocity
+
+    # for loop iterate and plot abundance of each element at a time
+    fig = go.Figure()
+    for index, row in df.T.iterrows():
```

Comment:
Are you sure this is the most time-efficient way to iterate over a pandas dataframe?